notebooks/README.md

```diff
@@ -1,24 +1,49 @@
-Somewhat experimental setup for tracking canonical versions of Learn notebooks in version control.
+The 'notebook pipeline' in this directory is a pseudo templating system with kaggle kernels API integration.
 
-`partials/python/` contains 'templates' for all the notebooks in the Python track (with `tut_1.ipynb` being the first tutorial notebook, `ex_1.ipynb` being the corresponding exercise notebook, etc.). This is where notebook editing should happen.
+## Directory structure
 
-These notebooks contain some 'macros' in the source of some cells, such as `#$END_OF_EXERCISE$` or `#$EXERCISE_URL(4)$`. These get replaced (see `lesson_preprocessor.py`) when rendering the final notebook. To render some notebooks (writing them to the `rendered/` subdirectory), use `render.sh`. e.g.
+To start a new track (i.e. a themed sequence of notebooks such as are found on the Kaggle Learn homepage, or in Kaggle Learn Challenges) called "spam", run `new_track.sh spam`. This will create a `spam/` subdirectory, with the following contents:
 
-    render.sh partials/python/*.ipynb
+```
+partials/
+rendered/
+pushables/
+__init__.py
+track_meta.py
+track_config.yaml
+```
 
-When you're ready to sync changes to the site, run `prepare_push.py`, passing the name of the track to prepare (e.g. `prepare_push.py python`). This will create a directory structure in a subdirectory `pushables` which is amenable to the Kernels API.
+`partials/` is where the notebooks you author should go. These will be straightforward ipynb notebooks which (perhaps with some setup such as downloading necessary datasets, installing libraries, and careful path manipulation) may be runnable locally. However, they're *also* recipes for generating ipynb notebooks.
 
-To push a particular notebook, do something like...
+`rendered/` is where the notebooks generated from the `partials/` recipes go. These are what get synced to Kaggle Kernels. You should not edit these directly (treat them as build products).
 
-    cd pushables/python/ex_4
-    kaggle k push -p .
+`pushables/` contains `kernel-metadata.json` files for syncing notebooks to Kaggle Kernels via the API.
 
-(NB: Should be possible to do in just one step like...
+`track_meta.py` defines some metadata, mostly about the notebooks you'll be syncing. A notebook in the `partials/` subdirectory will only be rendered (and have a `kernel-metadata.json` file generated) if it has an entry in `track_meta.py`. (This means you're welcome to put as many throwaway testing notebooks as you like in `partials` without worrying about them breaking anything.)
+See `examples/example_track/track_meta.py` for an exhaustively commented example.
 
-    kaggle k push -p pushables/python/ex_4
-    
-But API currently has some issues when run on a directory other than pwd.)
+`track_config.yaml` is much shorter. Whereas `track_meta.py` deals with the "what", this deals with the "how". 
```

Comment:
Given how short track_config is, I'd prefer just adding it to track_meta.py

Also, the type of information I see in these doesn't seem overwhelmingly different.

Reply:
The difference is subtle but, I think, important. `track_meta.py` describes the content of the track. It will generally only change when the content changes (adding a notebook, changing a lesson topic, etc.). `track_config.yaml` describes how you want to generate the track's notebooks at the moment. It will change depending on what you're trying to get done - whether you're debugging, staging, or publishing. 

Also I expect the number of fields in track_config to grow. For example, I think the 'development' flag will eventually split into some more granular controls. One determining whether to lead with the pip install trick, another determining whether to leave in cells marked as 'debug', one saying whether to push to a new, throwaway kernel id (to test some change in Kernels before affecting the 'live' version that users see), etc.

Reply:
Actually, thinking about it some more, it's probably more accurate to think of this as a 'build config'. And maybe the options should be overrideable via command line flags to render.py/prepare_push.py? I think this abstraction will probably evolve once I've put in some more hours using the system. (That's basically how I got here from version 0 of the pipeline that I was using with the Python challenge/track)

Reply:
I hope we're careful about adding too much flexibility/complexity. Each additional flag increases the cognitive load of using the pipeline... and there's likely a long tail of things we'll want for once but won't want again (especially since we're not releasing new tracks incredibly frequently.)